tests/test-org-ql.el

```diff
@@ -2111,7 +2111,84 @@ with keyword arg NOW in PLIST."
           (it "Refuses to link to non-file-backed buffer"
             (expect (var-after-link-save-open 'org-ql-view-buffers-files link-buffer query
                                               :buffer link-buffer)
-                    :to-throw 'user-error '("Views that search non-file-backed buffers can’t be linked to"))))))
+                    :to-throw 'user-error '("Views that search non-file-backed buffers can’t be linked to"))))
+        (describe "Completion for Files/buffers"
+          (describe "Contracting org-ql-view-buffers-files"
+            (it "org-agenda-files from list"
+              (spy-on 'org-agenda-files :and-return-value temp-filenames)
+              (expect (org-ql-view--contract-buffers-files temp-filenames) :to-equal "org-agenda-files"))
+            (it "org-directory"
+              (spy-on 'org-ql-search-directories-files :and-return-value temp-filenames)
+              (spy-on 'org-agenda-files :and-return-value '())
+              (expect (org-ql-view--contract-buffers-files temp-filenames) :to-equal temp-filenames))
+            (it "buffer"
+              (with-current-buffer (org-ql-test-data-buffer "data.org")
+                (expect (org-ql-view--contract-buffers-files (current-buffer)) :to-equal "buffer")))
+            (it "org-agenda-files from symbol"
+              (spy-on 'org-agenda-files :and-return-value temp-filenames)
+              (expect (org-ql-view--contract-buffers-files 'org-agenda-files) :to-equal "org-agenda-files")
+              (expect (org-ql-view--contract-buffers-files #'org-agenda-files) :to-equal "org-agenda-files"))
+            (it "Non specific list"
+              (let ((value1 '("a.org" "b.org"))
+                    (value2 'a))
+                (expect (org-ql-view--contract-buffers-files value1) :to-equal value1)
+                (expect (org-ql-view--contract-buffers-files value2) :to-equal value2))))
+          (describe "Expanding org-ql-view-buffers-files"
+            (it "all"
+              (let* ((buffer-names (list (make-temp-file "test" nil ".org") (make-temp-file "test" nil ".other")))
+                     (buffers (mapcar #'get-buffer-create buffer-names)))
+                (save-excursion
+                  (switch-to-buffer (car buffers))
+                  (org-mode))
+                (spy-on 'buffer-list :and-return-value buffers)
+                (expect (org-ql-view--expand-buffers-files "all") :to-equal (list (car buffers)))))
+            (it "org-agenda-files"
+              (spy-on 'org-agenda-files :and-return-value "value for org-agenda-files")
```

Comment:
Mocking the function's value is probably fine, but it might be good to use a list value rather than a string, since that's what it's expected to return.

Reply:
using let instead of spy